app/code/local/Dwyera/Pinpay/Model/PaymentMethod.php

```diff
@@ -230,9 +339,16 @@ protected function _buildRequest($payment, $amount, $email)
         $request->setEmail($email)->
             setAmount($request::getAmountInCents($amount))->
             setDescription("Quote #:" . $payment->getOrder()->getRealOrderId())->
-            setCardToken($payment->getAdditionalInformation('card_token'))->
             setIpAddress($payment->getAdditionalInformation('ip_address'));
 
+        //Consider card token only if the customer token is not present. 
+        if($payment->getAdditionalInformation("customer_token")!=""){
+            $request->setCustomerToken($payment->getAdditionalInformation("customer_token"));
+        }else{
+            $request->setCardToken($payment->getAdditionalInformation("card_token"));
```

Comment:
@zainengineer Yes if customer_token doesnot exist then there is always card token that we could use to make payment. But this happens only when customer doesnot exist.